app/gui2/src/util/qualifiedName.ts

```diff
@@ -30,7 +31,11 @@ export function isQualifiedName(str: string): str is QualifiedName {
 }
 
 export function tryQualifiedName(str: string): Result<QualifiedName> {
-  return isQualifiedName(str) ? Ok(str) : Err(`"${str}" is not a valid qualified name`)
+  if (isQualifiedName(str)) {
+    return Ok(str.replace(mainSegmentRegex, '$1') as QualifiedName)
+  } else {
+    return Err(`"${str}" is not a valid qualified name`)
+  }
```

Comment:
Now we modify the received string if it contains `Main` module.

Reply:
The problem is that the function does not tell about _fully_ qualified names (in contrast to the old GUI). Maybe it should - I don't remember any usage with partial qualified names. But then, it requires a bit of refactoring here (identifier is no longer a valid QN). 

The simplest solution would be to have a function "normalize FQN" and do not discern (un)normalized FQNs from QNs on the type level yet.

Reply:
Yep, it was my concern, especially when having functions like `qnJoin` which inherently works with partially qualified names. 

The problem with `normalize FQN` function is that it‘s not clear when it should be applied, and we can accidentally skip places where normalization is needed. I think it is slightly better then my current solution, though.

Reply:
Added `normalizeQn` function